[docs][NFC] Add usage instructions for FatLTO

ilovepi opened the pull request:
We included these in the commit message when we added `-ffat-lto-objects`, but they should be in the documentation as well.

ilovepi committed:
[docs] Add usage instructions for FatLTO

We included these in the commit message when we added
`-ffat-lto-objects`, but they should be in the documentation as well.

petrhosek commented on llvm/docs/FatLTO.rst:
```diff
@@ -76,3 +76,56 @@ Supported File Formats
+
+.. code-block:: console
+
+   $ clang -fno-lto -ffat-lto-objects -fuse-ld=lld foo.c
```
This is just a nit, but would it be possible to use `example.c` instead of `foo.c` here and below?

nikic commented on llvm/docs/FatLTO.rst:
```diff
@@ -76,3 +76,56 @@ Supported File Formats
+no effect.
+
+
+Compile and link. Use the object code from the fat object without LTO.
```
I'm not sure these "compile and link" examples make a lot of sense. I think fat LTO is only meaningful if compile and link is separate.

ilovepi replied:
Yeah, I'll revise the examples to have a common first compile step and then different link steps.

nikic commented on llvm/docs/FatLTO.rst:
```diff
@@ -76,3 +76,56 @@ Supported File Formats
+
+.. code-block:: console
+
+   $ clang -flto=thin -ffat-lto-objects -funified-lto -fuse-ld=lld foo.c
```
Isn't `-funified-lto` the default/required for fat LTO now? I'd expect the flag to have no additional effect.

ilovepi replied:
Ah, good point. Thanks.

ilovepi committed:
!fixup Address comments

Rename foo.c to example.c

Remove UnifiedLTO examples, since FatLTO always uses that pipeline now.

ilovepi committed:
!fixup Focus on separate linking in the example

MaskRay commented on llvm/docs/FatLTO.rst:
```diff
@@ -76,3 +76,40 @@ Supported File Formats
+
+.. code-block:: console
+
+   $ clang -fno-lto -ffat-lto-objects -fuse-ld=lld example.o
```
The description can state that `-ffat-lto-objects` is a no-op for linking when `-fno-lto` is specified.

ilovepi replied:
done.

MaskRay commented on llvm/docs/FatLTO.rst:
```diff
@@ -76,3 +76,40 @@ Supported File Formats
+Usage
+===========
+
+Users are expected to pass ``-ffat-lto-objects`` to clang in addition to one of
```
Clang users can specify `-ffat-lto-objects` along with `-flto` or `-flto=thin`. Without the ``-flto`` flag, ``-ffat-lto-objects`` has no effect.

ilovepi replied:
done

ilovepi committed:
!fixup Update text with reviewer suggestions.

ilovepi commented on llvm/docs/FatLTO.rst:
```diff
@@ -76,3 +76,40 @@ Supported File Formats
+
+.. code-block:: console
+
+   $ clang -flto -ffat-lto-objects example.c -o example.o
```
Good catch. Thank you!
